packages/node_modules/@ciscospark/internal-plugin-ediscovery/src/ediscovery.js

```diff
@@ -0,0 +1,520 @@
+/* eslint-disable prefer-template */
+/*!
+ * Copyright (c) 2015-2017 Cisco Systems, Inc. See LICENSE file.
+ */
+
+import {waitForValue, SparkPlugin} from '@ciscospark/spark-core';
+import reportRequest from './report-request';
+import ReportGenerator from './report-generator';
+
+/**
+ * @class Ediscovery is used by compliance officers to run compliance reports
+ *
+ */
+const Ediscovery = SparkPlugin.extend({
+  namespace: 'Ediscovery',
+
+  /**
+   * Creates a compliance report with a specific set of search parameters
+   *
+   * @param {String} name - A label to identify the report
+   * @param {String} description - A textual summary of the reports purpose
+   * @param {Array<String>} emails - A list of user emails relevant to the report
+   * @param {Array<String>} keywords - A list of search terms relevant to the report
+   * @param {Array<String>} spaceNames - A list of space names relevant to the report
+   * @param {Array<String>} spaceIds - A list of space ids relevant to the report
+   * @param {Object} range - Contains the start time and end time defining the search period
+   * @returns {Promise<ResponseEntity>} Http response containing the new report record
+   */
+  @waitForValue('@')
+  createReport(name, description, emails, keywords, spaceNames, spaceIds, range) {
+    if (!emails || !keywords || !spaceNames || !spaceIds || !range) {
+      throw Error('Invalid parameter(s)');
+    }
+
+    const self = this;
+    const encryptionPromise = self._encryptRequestParams(name, description, keywords);
+    const uuidPromise = self._asUUIDs(emails);
+
+    return Promise.all([encryptionPromise, uuidPromise])
+      .then((values) => {
+        const body = reportRequest;
+        body.name = values[0].encryptedName;
```

Comment:
I don't think you want to assign `body` to `reportRequest` and then modify it. You will end up changing `reportRequest` by reference. Here is an example

```
const report = {
	a: 'foo',
	b: 'bar'
}

const body = report;

body.a = 'bar';

console.log('body', body);
console.log('report', report);
```

yields
```
body { a: 'bar', b: 'bar' }
report { a: 'bar', b: 'bar' }
```

Reply:
I think we should make the reportRequest into a proper class, use it for the function param and create another for this body object.